File: hyoid_app/lib/features/doctor/data/models/prescription.dart
class Medicine {
  final String name;
  final String dosage;
  final String duration;

  Medicine({
    required this.name,
    required this.dosage,
    required this.duration,
  });

  Map<String, dynamic> toJson() {
    return {
      'name': name,
      'dosage': dosage,
      'duration': duration,
    };
  }

  factory Medicine.fromJson(Map<String, dynamic> json) {
    return Medicine(
      name: json['name'] ?? '',
      dosage: json['dosage'] ?? '',
      duration: json['duration'] ?? '',
    );
  }
}

class Prescription {
  final List<Medicine> medicines;
  final String notes;

  Prescription({
    required this.medicines,
    required this.notes,
  });

  Map<String, dynamic> toJson() {
    return {
      'medicines': medicines.map((m) => m.toJson()).toList(),
      'notes': notes,
    };
  }

  factory Prescription.fromJson(Map<String, dynamic> json) {
    return Prescription(
      medicines: (json['medicines'] as List?)
              ?.map((m) => Medicine.fromJson(m))
              .toList() ??
          [],
      notes: json['notes'] ?? '',
    );
  }
}

File: hyoid_server/models/Prescription.js
const mongoose = require('mongoose');

const prescriptionItemSchema = new mongoose.Schema({
  medicineName: {
    type: String,
    required: true,
  },
  dosage: {
    type: String,
    required: true,
  },
  frequency: {
    type: String,
    required: true,
  },
  duration: {
    type: Number, // in days
    required: true,
  },
  instructions: {
    type: String,
    required: true,
  },
  genericAlternative: {
    type: String,
  },
});

const prescriptionSchema = new mongoose.Schema({
  consultationId: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'Consultation',
    required: true,
  },
  doctorId: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'Doctor',
    required: true,
  },
  patientId: {
    type: String,
    required: true,
  },
  issuedAt: {
    type: Date,
    default: Date.now,
  },
  diagnosis: {
    type: String,
    required: true,
  },
  medicines: [prescriptionItemSchema],
  notes: {
    type: String,
  },
  doctorSignature: {
    type: String,
    required: true,
  },
  isDigital: {
    type: Boolean,
    default: true,
  },
}, {
  timestamps: true,
});

module.exports = mongoose.model('Prescription', prescriptionSchema);